plugin/src/config.ts

```diff
@@ -0,0 +1,108 @@
+import { ExpoConfig } from '@expo/config-types';
+
+export type ExpoConfigFacebook = Pick<
+  ExpoConfig,
+  | 'facebookScheme'
+  | 'facebookAdvertiserIDCollectionEnabled'
+  | 'facebookAppId'
+  | 'facebookAutoInitEnabled'
+  | 'facebookAutoLogAppEventsEnabled'
+  | 'facebookDisplayName'
+>;
+
+export type ConfigProps = {
+  /**
+   * Used for all Facebook libraries. Set up your Facebook App ID at https://developers.facebook.com.
+   */
+  appID?: string;
+  /**
+   * Used for native Facebook login.
+   */
+  displayName?: string;
+  /**
+   * Used for Facebook native login. Starts with 'fb' and followed by a string of digits, like 'fb1234567890'. You can find your scheme [here](https://developers.facebook.com/docs/facebook-login/ios)in the 'Configuring Your info.plist' section (only applicable to standalone apps and custom Expo Go apps).
+   */
+  scheme?: string;
+
+  clientToken?: string;
```

Comment:
I think it is important to at least mention that you pull this from the facebook developer dashboard as well, and that it will be required in the next major version of the facebook SDKs so you should configure it even if it is temporarily optional.

Thinking further: many of these ConfigProps are marked as optional, but, the appID and displayName and scheme etc - some of those are required and not optional, aren't they?

Reply:
I make some config are optional because I want to support Expo User moving from `expo-facebook` to this package. They ready has some config in the app.json so they don't have to declare it twice. But, I will add validation for missing `appID` and `displayName`. (`scheme` will default to `fb` + appID).

```
    facebookAppId,
    facebookDisplayName,
    facebookScheme,
    facebookAutoInitEnabled,
    facebookAutoLogAppEventsEnabled,
    facebookAdvertiserIDCollectionEnabled,
  ```